pandas/core/indexing.py

```diff
@@ -213,6 +214,16 @@ def _setitem_with_indexer(self, indexer, value):
                 val = list(value.values()) if isinstance(value,dict) else value
                 take_split_path = not blk._can_hold_element(val)
 
+        if isinstance(indexer, tuple) and len(indexer) == len(self.obj.axes):
+
+            for i, ax in zip(indexer, self.obj.axes):
+
+                # if we have any multi-indexes that have non-trivial slices (not null slices)
+                # then we must take the split path, xref GH 10360
+                if isinstance(ax, MultiIndex) and not (is_integer(i) or is_null_slice(i)):
+                    take_split_path = True
+                    break
+
```

Comment:
@jreback looks like this is intended for Panel, but is still reached.  Any chance it is unnecessary?  Or more specifically, I'd like to rule out the possibility of `take_split_path = True` being set on 224 for `Series`.  Is that unreachable for some reason?

Reply:
umm maybe, the whole take_split_path is based on the notion that we have a block  of a single type and we are indexing into it for smaller number of columns than the whole block so we *might* need to split it before we index. fairly dense & complicated, but necessary in some way.

to answer your question though, I agree you might be able to remove this entire clause.

Reply:
Apparently there was still a use case: https://github.com/pandas-dev/pandas/pull/27855